snapcraft/internal/project_loader/_config.py

```diff
@@ -294,3 +395,50 @@ def _expand_filesets_for(step, properties):
             new_step_set.append(item)
 
     return new_step_set
+
+
+class _Architecture:
+    def __init__(self, *, build_on, run_on=None):
+        self.build_on = build_on
+        if isinstance(build_on, str):
```

Comment:
why double assign when you can make it clearer with an else?
```yaml
if isinstance(build_on, str):
    self.build_on = [build_on]
else:
    self.build_on = build_on
```

same for `run_on`

Reply:
Fixed.